github/transport_test.go

```diff
@@ -45,6 +45,39 @@ func TestEtagTransport(t *testing.T) {
 	}
 }
 
+func TestUserAgentTransport(t *testing.T) {
```

Comment:
I'd love to have some kind of integration test for REST and GraphQL that shows that the header is actually set, but I couldn't find a super natural place to do that. Any ideas?

Reply:
Hmm...were we to do that, we'd need actual credentials querying an actual repo. I think both adding a new file or expanding the current transport tests would be appropriate. 

Docs about writing Terraform acceptance tests are [here](https://www.terraform.io/plugin/framework/acctests) or look at existing tests [here](https://github.com/integrations/terraform-provider-github/blob/main/github/resource_github_branch_protection_test.go#L13) as a model. 

Let me know if you'd like to pair on this!